test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/integration/ProcessInstanceMigrationExternalDatabaseIntegrationTest.java

```diff
@@ -0,0 +1,210 @@
+
+    private static final String PROCESS_ID = "test.myprocess";
+
+    private final ObjectMapper mapper = new ObjectMapper().findAndRegisterModules();
+
+    protected KieServicesClient kieServicesClient;
+    protected ProcessServicesClient processServicesClient;
+    protected String kieServerId;
+    protected HttpClient client;
+    protected List<Long> instancesIds = new ArrayList<>();
+
+
+    @BeforeClass
+    public static void initializeDeployment() {
+        Assume.assumeTrue(isExternalDatabaseAllocated());
+        try {
+            deploymentScenario = deploymentScenarioFactory.getKieServerWithExternalDatabaseScenarioBuilder()
+                                                          .withInternalMavenRepo(true)
+                                                          .withContainerDeployment(KIE_CONTAINER_DEPLOYMENTS)
+                                                          .withProcessMigrationDeployment()
+                                                          .build();
+
+
+            KjarDeploymentScenarioListener.addKjarDeployment(deploymentScenario, OLD_KJAR, NEW_KJAR);
+        } catch (UnsupportedOperationException ex) {
+            Assume.assumeFalse(ex.getMessage().startsWith("Not supported"));
+        }
+        deploymentScenario.setLogFolderName(ProcessInstanceMigrationExternalDatabaseIntegrationTest.class.getSimpleName());
+        ScenarioDeployer.deployScenario(deploymentScenario);
+
+    }
+
+    @Before
+    public void setUp() throws KeyManagementException, NoSuchAlgorithmException, KeyStoreException {
+        logger.info("URL " + deploymentScenario.getKieServerDeployment().getUrl());
+        kieServicesClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
+        processServicesClient = KieServerClientProvider.getProcessClient(deploymentScenario.getKieServerDeployment());
+
+        kieServerId = kieServicesClient.getServerInfo().getResult().getServerId();
+        client = HttpClientBuilder.create().setDefaultCredentialsProvider(getBasicAuth()).build();
+    }
+
+    @Test
+    public void testProcessInstanceMigrationIntegration() {
+        // Given
+        startProcesses();
+
+        // When
+        try {
+            createMigration();
+        } catch (IOException | JAXBException e) {
+            throw new RuntimeException("Process instance migration failed", e);
+        }
+
+        // Then
+        List<ProcessInstance> instances = processServicesClient.findProcessInstances(OLD_CONTAINER_ID, 0, 10);
+        Assertions.assertThat(instances).hasSize(1);
+        Assertions.assertThat(instances.get(0).getId()).isIn(instancesIds);
+
+        instances = processServicesClient.findProcessInstances(NEW_CONTAINER_ID, 0, 10);
+        Assertions.assertThat(instances).hasSize(1);
+        Assertions.assertThat(instances.get(0).getId()).isIn(instancesIds);
```

Comment:
Sorry if this is a dummy question, but if I start the same process twice, I would have expected to have 0 instances of old container and two instances of the new container. What am I missing?

Reply:
One of the process instances was migrated. That is purposed of this test :)